common/src/main/java/org/jboss/eap/qe/ts/common/docker/Docker.java

```diff
@@ -0,0 +1,188 @@
+ *
+ * // Docker container will be started in @BeforeClass phase - see list and current configuration of current {@link DockerContainers}
+ * @ClassRule
+ * public static Docker jaegerContainer=DockerContainers.jaeger();
+ *
+ *
+ * @Test
+ * public void test()  {
+ *  // test something with docker container...
+ * }
+ * }
+ */
+public class Docker extends ExternalResource {
+    private final String uuid;
+    private final String name;
+    private final String image;
+    private final List<String> ports = new ArrayList<>();
+    private final Map<String, String> environmentVariables = new HashMap<>();
+    private final List<String> commandArguments = new ArrayList<>();
+    private String awaitedLogLine;
+    private long waitTimeoutInMillis = 600_000; // 10 minutes
+
+    private final ExecutorService outputPrinter = Executors.newSingleThreadExecutor();
+
+    public Docker(String name, String image) {
+        this.uuid = UUID.randomUUID().toString();
+        this.name = name;
+        this.image = image;
+    }
+
+    public Docker waitForLogLine(String logWait) {
+        this.awaitedLogLine = logWait;
+        return this;
+    }
+
+    public Docker waitTimeout(long timeoutInMillis) {
+        this.waitTimeoutInMillis = timeoutInMillis;
+        return this;
+    }
+
+    public Docker port(String port) {
+        this.ports.add(port);
+        return this;
+    }
+
+    public Docker envVar(String key, String value) {
+        environmentVariables.put(key, value);
+        return this;
+    }
+
+    public Docker cmdArg(String commandArgument) {
+        commandArguments.add(commandArgument);
+        return this;
+    }
+
+    private void start() throws IOException, TimeoutException, InterruptedException {
+        List<String> cmd = new ArrayList<>();
+
+        cmd.add("docker");
+        cmd.add("run");
+        cmd.add("--name");
+        cmd.add(uuid);
+
+        for (String port : ports) {
+            cmd.add("-p");
+            cmd.add(port);
+        }
+
+        for (Map.Entry<String, String> envVar : environmentVariables.entrySet()) {
+            cmd.add("-e");
+            cmd.add(envVar.getKey() + "=" + envVar.getValue());
+        }
+
+        cmd.add(image);
+
+        cmd.addAll(commandArguments);
+
+        System.out.println(Ansi.ansi().reset().a("Starting container ").fgCyan().a(name).reset()
+                .a(" with ID ").fgYellow().a(uuid).reset());
+
+        Process dockerRunProcess = new ProcessBuilder()
+                .redirectErrorStream(true)
+                .command(cmd)
+                .start();
+
+        CountDownLatch latch = new CountDownLatch(1);
+
+        outputPrinter.execute(() -> {
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(dockerRunProcess.getInputStream()))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    System.out.println(Ansi.ansi().fgCyan().a(name).reset().a("> ").a(line));
```

Comment:
`System.out` again.

Reply:
Correct, we'll need to add some logger which would be used in the whole test suite. Any preference/suggestion here? 

What about to use lombok `@Log4j` - https://projectlombok.org/features/log

Reply:
I didn't even know that Lombok project exists. Looks good - I'd like to use it in the testsuite. Personally, I prefer Slf4j, but that's just a matter of habit. Thank you for the tip!

Reply:
I have no problem with Slf4j as well. I'll bring this topic on chat with others.